lib/util/http/retry-after.ts

```diff
@@ -0,0 +1,93 @@
+import { setTimeout } from 'timers/promises';
+import { RequestError } from 'got';
+import { logger } from '../../logger';
+import * as hostRules from '../host-rules';
+import { parseUrl } from '../url';
+import type { Task } from './types';
+
+const hostBlocks = new Map<string, Promise<void>>();
+
+const maxRetries = 2;
+
+export async function wrapWithRetry<T>(
+  url: string,
+  task: Task<T>,
+  getDelaySeconds: (err: unknown) => number | null,
+  maxRetryAfter = 60,
+): Promise<T> {
+  const key = parseUrl(url)?.host ?? /* istanbul ignore next */ url;
+
+  let retries = 0;
+  for (;;) {
+    try {
+      const delayPromise = hostBlocks.get(key);
+      if (delayPromise) {
+        await delayPromise;
+        hostBlocks.delete(key);
+      }
+
+      return await task();
```

Comment:
we need to consider that got is doing retry too by default, we disabled it for tests. so we need to ensure that got doesn't retry when we get that retry after header

Reply:
As far as I know, it doesn't. From what I've seen in its repo, it turns off any automatic retry logic for this header.

Reply:
did you check the code for got v11?

https://github.com/sindresorhus/got/tree/v11.8.6#retry

looks like got automatically retries on 429 with retry-after header. at least if timeout isn't reached.

https://github.com/sindresorhus/got/blob/v11.8.6/source/core/index.ts#L2551

it seems got is retrying

Reply:
So we need to distinguish between a timeout (which I generally associate with a connection "hanging") versus when we get an almost immediately response but it's a 429.

@zharinov can you do some more local testing with a mock http server and got?

Reply:
Yes, you're right, it's retrying on its own

Reply:
Now it's disabled